### PR TITLE
Add github actions for go fmt, test

### PR DESCRIPTION
**Pull request by mnm678**
Solves #16

### CODE REVIEW

File: go.mod
Comment (trishankatdatadog) [inline go.mod:5]: Hmm, do we need a `go.mod` here?

Comment (mnm678) [reply go.mod:5]: oh, this was messing up go fmt, but I guess it does belong in a different pr

**Comment by mnm678** [comment]
Can I get a review/approval @sudo-bmitch, @trishankatdatadog? 

You can see the workflow in action here: https://github.com/mnm678/tuf-1/pull/6. It fails on newer go versions, which I'll fix in another pr.

**Review by trishankatdatadog — APPROVED** [review]

**Comment by sudo-bmitch** [comment]
Belated review from me, but I also noticed that the actions tab didn't populate, so it got my attention.

I think the path needs to be `.github/workflows/` (plural). And since the action won't modify the commit, running `go fmt` in the action isn't much value, we just need to require that from committers before merging PR's. Most of us do that in our IDEs automatically on save, but we could include a Makefile that does this before locally building and testing.

I can probably get a PR with these later this week since they're pretty standard in my projects.

**Comment by trishankatdatadog** [comment]
> I think the path needs to be `.github/workflows/` (plural). And since the action won't modify the commit, running `go fmt` in the action isn't much value, we just need to require that from committers before merging PR's. Most of us do that in our IDEs automatically on save, but we could include a Makefile that does this before locally building and testing.

`go fmt` here could at least help detect and prevent issues, no?

**Comment by mnm678** [comment]
Thanks @sudo-bmitch, I opened #23 to add the s, but it still doesn't seem to run the action?

My thought was that the `go fmt` run would mostly verify that the committer ran that correctly, but a Makefile that fixes it would be even better.

**Comment by sudo-bmitch** [comment]
@trishankatdatadog 

> `go fmt` here could at least help detect and prevent issues, no?

That would likely go ignored, the return code is 0 even when formatting needs to be fixed. At most you see a filename that was updated in the output, which requires someone to double check that output on every run. Unless we're doing something I'm missing to ensure there's no files changed by `go fmt`.

**Pull request merged**
